Sources/ArgumentParser/Completions/BashCompletionsGenerator.swift

```diff
@@ -187,18 +187,44 @@ extension ArgumentDefinition {
   }
 
   /// Returns the bash completions that can follow this argument's `--name`.
+  ///
+  /// Uses bash-completion for file and directory values if available.
   fileprivate func bashValueCompletion(_ commands: [ParsableCommand.Type]) -> String {
     switch completion.kind {
     case .default:
       return ""
       
-    case .file(_):
-      // TODO: Use '_filedir' when available
-      // FIXME: Use the extensions array
-      return #"COMPREPLY=( $(compgen -f -- "$cur") )"#
+    case .file(let extensions) where extensions.isEmpty:
+      return """
+        if declare -F _filedir >/dev/null; then
+          _filedir
+        else
+          COMPREPLY=( $(compgen -f -- "$cur") )
+        fi
+        """
+
+    case .file(let extensions):
+      var safeExts = extensions.map { String($0.flatMap { $0 == "'" ? ["\\", "'"] : [$0] }) }
+      safeExts.append(contentsOf: safeExts.map { $0.uppercased() })
+      
+      return """
+        if declare -F _filedir >/dev/null; then
+          \(safeExts.map { "_filedir '\($0)'" }.joined(separator:"\n  "))
```

Comment:
I think we need an additional call to `_filedir -d` here – when I run completions on an extension-constrained `.file` item, it should show me the matching files _plus_ the available directories, since I might need to navigate before finding the correct file.

Reply:
```
bash-3.2$ math stats quantiles --file 
CHANGELOG.md        CMakeLists.txt      CODE_OF_CONDUCT.md  CONTRIBUTING.md     LICENSE.txt         README.md           
bash-3.2$ math stats quantiles --file Tes
<no output>
bash-3.2$ math stats quantiles --file Tests/
<auto-completes to:>
bash-3.2$ math stats quantiles --file Tests/CMakeLists.txt
```

Reply:
Good catch, done and done! 👍